### PR TITLE
[1.1.0] I've created a validation method on \Raml\NamedParameter

### PR DESCRIPTION
**Pull request by CozyLife**
\Raml\NamedParameter::validate() allows for quick validation of a named parameter without having to individually test everything.

### CODE REVIEW

File: src/NamedParameter.php [hunk 1]
```diff
@@ -669,4 +672,151 @@ public function setDefault($default)
...
+    public function validate($param)
+    {
+        /**
+         * If we don't have a value to validate, check if it's required.
+         *
+         * @link http://raml.org/spec.html#required
+         */
+        if (in_array($param, array(null, ''), true)) {
+            if ($this->isRequired()) {
+                throw new \Exception($this->getKey().' is required', 1);
```
Comment (alecsammon) [inline src/NamedParameter.php]: We are currently in the process of migrating default Exceptions into custom exceptions. This makes it much easier when using the library to catch a specific type of exception. There already is a ValidationExceptionInterface, just haven't got around to having any exception extend it. Might be a good time to do so.

File: src/NamedParameter.php [hunk 2]
```diff
@@ -669,4 +672,151 @@ public function setDefault($default)
...
+                if (!is_string($param)) {
+                    throw new \Exception($this->getKey().' is not a string', 2);
+                }
+    
+                /**
+                 * Check the length of a string.
+                 *
+                 * @link http://raml.org/spec.html#minlength
+                 */
+                if (!empty($min_len = $this->getMinLength()) && strlen($param) < $min_len) {
```
Comment (alecsammon) [inline src/NamedParameter.php]: Expressions in empty() are php 5.5 and greater only. We need to support 5.4 also

Also - personally I also don't like assignments in conditions. I think they can be confusing

File: src/NamedParameter.php (hunk 1 above)
Comment (alecsammon) [inline src/NamedParameter.php]: May also be a good way of documenting what the different exception codes are.

File: src/NamedParameter.php [hunk 3]
```diff
@@ -386,11 +386,14 @@ public function setEnum(array $enum)
     /**
      * Get the pattern regular expression
      *
+     * @param boolean $typeCheck Set to false to only return the "pattern" value from the RAML data (Defaults to
+     *                           true for backwards compatability to return type checking strings when a validation
+     *                           pattern is not present in the RAML data)
      * @return string
      */
-    public function getValidationPattern()
+    public function getValidationPattern($typeCheck = true)
```
Comment (alecsammon) [inline src/NamedParameter.php]: See comment on other pull request.

File: src/NamedParameter.php [hunk 4]
```diff
@@ -669,4 +672,151 @@ public function setDefault($default)
...
+                // No break
+    
+            case 'integer':
+    
+                /**
+                 * Integers must be of type integer.
+                 *
+                 * @link http://raml.org/spec.html#type
+                 */
+                if ($this->getType() === 'integer' && !is_int($param)) {
```
Comment (alecsammon) [inline src/NamedParameter.php]: If you swapped the case 'number' and case 'integer' around - then you wouldn't need this extra check for type integer. As all integers should be numbers.

i.e. 
```
case 'integer'
  if (!is_int($param)) {
    throw new \Exception();
  }
  // no break
case 'number':
  if(!is_numeric($param)) {
  ...
```

File: src/NamedParameter.php [hunk 5]
```diff
@@ -669,4 +672,151 @@ public function setDefault($default)
...
+                throw new \Exception($this->getKey().' is required', 1);
+            }
+                
+            return;
+                
+        }
+        
+        switch($this->getType()) {
+            
+            case 'boolean':
```
Comment (alecsammon) [inline src/NamedParameter.php]: we already have constants for these strings - might be nicer to use them

```
case self::TYPE_BOOLEAN:
```

File: src/NamedParameter.php (hunk 2 above)
Comment (alecsammon) [inline src/NamedParameter.php]:
```
                if ($this->getMinLength() !== null && strlen($param) < $this->getMinLength()) {
                    throw new \Exception($this->getKey().' must be at least '.$this->getMinLength().' characters long', 3);
                }
```

**Comment by alecsammon** [comment]
Hi Rick. Thanks for this pull request. It's something I've been meaning to write for some time now, but just not had the chance. 

Sorry - have a number of comments (as always), however please don't treat it like a dictatorship on my behalf, more of a conversation to ensure we get the best result, feel free to disagree!!

The tests on this pull request are breaking on 5.4 - the fix is in the comments above.

Thank you again for all your contributions.

**Comment by CozyLife** [comment]
If I disagree, I make sure to disclose it, as you might have noticed in my original reaction to the SemVer conversation. ;) I'm glad I could help, and as far as I can tell you have valid points, so I'll make the changes.

The only part is have to disagree on is the switch versus omitting the RegEx. It wouldn't be backwards compatible and I forgot that it would have to bump the major version when I had originally agreed with it. Let me know how you feel about that change.

**Comment by CozyLife** [comment]
All clear.

**Comment by alecsammon** [comment]
Hi Rick, still got failing tests on 5.4 - looks like you can only have variables in empty(), not a method call.

**Comment by CozyLife** [comment]
They really run you through the gambit in 5.4, eh? Haha, alright, I'll fix that, too.

**Comment by CozyLife** [comment]
Alright, I've made empty() use only a variable. I've also changed the composer.json to pull your unofficial copy of json-schema.

**Pull request merged**